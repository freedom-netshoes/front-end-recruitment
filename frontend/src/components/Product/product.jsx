import React, { Component } from 'react'
import CSSModules from 'react-css-modules'
import { PropTypes } from 'prop-types'
import Installments from '../crossComponents/Installments/installments'
import Price from '../crossComponents/Price/price'
import Image from '../crossComponents/Image/image'
import Title from '../crossComponents/Title/title'
import styles from './product.less'

@CSSModules(styles)
export default class Product extends Component {
  static defaultProps = {
    title: '',
    sku: '',
    products: [],
    price: 0,
    installments: 0
  };

  static propTypes = {
    title: PropTypes.string,
    sku: PropTypes.string,
    products: PropTypes.array,
    price: PropTypes.number,
    installments: PropTypes.number
  };

  render () {
    const {
      title,
      sku,
      price,
      installments
    } = this.props

    return <div {... { title: title, styleName: 'product' }} >
      <Image {...{ sku: sku, title: title }} />
      <Title {...{ title: title }} />
      <div styleName='container'>
        <Price {...{ value: price, type: 'separated' }} />
        <Installments {...{ value: price, installments: installments }} />
      </div>
    </div>
  }
}
