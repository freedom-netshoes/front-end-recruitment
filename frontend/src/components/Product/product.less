.product {
    cursor: pointer;
    height: 350px;
    width: 230px;
    position: relative;
}

.container {
    position: absolute;
    bottom: 0;
    left: 50%;
    transform: translate(-50%, 0);
}