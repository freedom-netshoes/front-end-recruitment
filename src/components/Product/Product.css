
.item-list{
    display: block;
    width: 100%;
    max-width: 680px;
    text-align: left;
    margin: 0 auto;
}

.item{
    cursor: pointer;
    display: inline-block;
    width: 26.47%;
    margin-right: 6.4%;
    margin-top: 60px;
    font-size: 14px;
    vertical-align: top;
    text-align: center;
}
.item a{
    color: inherit;
    text-decoration: none;
}
.item .i img{
    width: 100%;
    height: auto;
}
.item .title{
    margin-top: 15px;
    font-size: 1em;
    line-height: 1em;
    color: #313135;
    font-weight: 
}
.item .title:after{
    content: "";
    width: 15px;
    height: 3px;
    background-color: #dfbd00;
    display: block;
    margin: 10px auto;
}
.item .price{
    font-size: 1.7em;
    color: #313135;
    text-decoration: none;
}
.item .price a{
    text-decoration: none;
}
.item .price span{
    font-weight: 700;
    text-decoration: none;
}
.item .price small{
    font-weight: 400;
    font-size: .5em;
}

.item .parceling{
    color: #999;
    font-weight: 600;
}
