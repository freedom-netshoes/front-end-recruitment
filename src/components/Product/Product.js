import React, { Component } from 'react'
import PropTypes from 'prop-types'
import imgProductDefault from './assets/img/img_1.jpg'

class Product extends Component {
    handleClick = () => {
        const { id, addToCart, removeFromCart, isInCart} = this.props;

        if (!isInCart) {
            addToCart(id);
        } else {
            removeFromCart(id);
        }
    }
    menuOpen = (e) => {
        e.stopPropagation()
        let wrapper = document.querySelector('.wrapper')
        this.handleClick()
        if (!wrapper.classList.contains('show-menu')) {
            wrapper.classList.add('show-menu');
        }
    }
    splitPrice = (valor) => {
        var decPart = (valor + "").split(".")[0];
        return decPart
    }
    splitDecimals = (valor) => {
        var decPart = (valor + "").split(".")[1];
        return decPart
    }
    returnPriceWithInstallments = (price, installments) => {
        var num = price / installments
        if (installments) {
            var n = num.toFixed(2);
            return n
        } else {
            return price
        }

    }

    render() {
        const { title, id, price, installments, currencyFormat } = this.props;

        return (
            <div onClick={this.menuOpen} key={id}>
                <a className="i card-link" title={title}>
                    <img src={imgProductDefault} alt={title} />
                </a>
                <div className="title">
                    <a className="n card-link">
                        <span>{title}</span>
                    </a>
                </div>
                <div className="price">
                    <a className="card-link">
                        <span className="full-mounted-price">
                            <span>
                                <small>{currencyFormat}</small>{this.splitPrice(price)},<small>{this.splitDecimals(price)}</small>
                            </span>
                        </span>
                    </a>
                </div>
                <a className="in parceling">ou {installments} x de {currencyFormat} {this.returnPriceWithInstallments(price,installments)}</a>
            </div>
        );
    }
}

Product.propTypes = {
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    price: PropTypes.number,
    currencyFormat: PropTypes.string,
    image: PropTypes.string,
    isInCart: PropTypes.bool.isRequired,
    addToCart: PropTypes.func.isRequired,
    removeFromCart: PropTypes.func.isRequired,
}

export default Product;
