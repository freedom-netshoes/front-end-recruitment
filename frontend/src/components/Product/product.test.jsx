/* global describe, it, expect */
import React from 'react'
import Product from './product'
import { shallow } from 'enzyme'

describe('Testing <Product /> component', () => {
  it('renders the Product component', () => {
    const component = shallow(<Product />)
    expect(component).toHaveLength(1)
  })

  it('Testing snapshop in Product ', () => {
    let testObj = {
      'id': 1,
      'sku': '18644119330491312',
      'title': 'Camisa Nike Corinthians II',
      'description': '14/15 s/nยบ',
      'availableSizes': ['S', 'G', 'GG', 'GGG'],
      'style': 'Preta com listras brancas',
      'price': 229.9,
      'installments': 9,
      'currencyId': 'BRL',
      'currencyFormat': 'R$',
      'isFreeShipping': true,
      'quantity': 2
    }
    let component = shallow(<Product {...{ testObj }} />)
    expect(component).toMatchSnapshot()
  })
})
