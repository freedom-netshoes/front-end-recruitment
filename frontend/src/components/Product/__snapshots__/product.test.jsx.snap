// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Product /> component Testing snapshop in Product  1`] = `
<div
  className="product"
  title=""
>
  <Image
    isThumb={false}
    sku=""
    title=""
  />
  <Title
    title=""
  />
  <div
    className="container"
  >
    <Price
      type="separated"
      value={0}
    />
    <Installments
      installments={0}
      type="simple"
      value={0}
    />
  </div>
</div>
`;
